src/test/ui/closures/2229_closure_analysis/migrations/insignificant_drop_attr.rs

```diff
@@ -0,0 +1,58 @@
+// run-pass
+
+#![deny(disjoint_capture_drop_reorder)]
+#![feature(rustc_attrs)]
+#![allow(unused)]
+
+#[rustc_insignificant_dtor]
+struct InsignificantDropPoint {
+    x: i32,
+    y: i32,
+}
+
+impl Drop for InsignificantDropPoint {
+    fn drop(&mut self) {}
+}
+
+// `String` implements drop which makes `Person` a droppable type.
+struct Person {
+    age: i32,
+    name: String,
+}
+
+// `TwoPeople` is also droppable as `Person` is. But since it is
+// annotated by `rustc_insignificant_dtor`, it should not be
+// treated as a significant drop and so drop reordering should
+// not raise an error due to precise capturing.
+#[rustc_insignificant_dtor]
+struct TwoPeople {
+    p1: Person,
+    p2: Person,
+}
+
+fn test1_all_need_migration() {
```

Comment:
describe what the test does, not the classes.

The function names need to be changed to reflect what is being tested